src/main/java/io/cdap/plugin/servicenow/apiclient/ServiceNowTableAPIClientImpl.java

```diff
@@ -362,8 +362,8 @@ public List<Map<String, Object>> fetchTableRecordsRetryableMode(String tableName
     try {
       retryer.call(fetchRecords);
     } catch (RetryException | ExecutionException e) {
-      LOG.error("Data Recovery failed for batch {} to {}.", offset,
-                (offset + limit));
+      LOG.error("Data Recovery failed for batch {} to {}.", offset, (offset + limit));
```

Comment:
Log and throw is an anti-pattern, it normally results in too many log messages. For example, something higher up the call stack catches this exception then logs it again, or decides to retry the method X times, resulting in X messages.

Instead, it is better to remove the log message and make sure the exception error message contains any relevant information.

Reply:
Removed the error logger. Added the error message in the exception. b450d01ecde6caac6e9e789958aff4c54b3e4cb0